channeldb/payments.go

```diff
@@ -251,9 +251,84 @@ type Payment struct {
 	Failure *FailureReason
 }
 
+// Route returns the route used to complete the payment, or the final route
+// attempted if the payment failed or is in-flight.
+func (p *Payment) Route() route.Route {
+	var rt route.Route
+	if p.Attempt != nil {
+		rt = p.Attempt.Route
+	}
+
+	return rt
+}
+
+// PaymentPreimage returns the preimage learned as a result of the payment if it
+// was successful, or the zero-preimage if the payment failed or is still
+// in-flight.
+func (p *Payment) PaymentPreimage() lntypes.Preimage {
+	var preimage lntypes.Preimage
+	if p.Preimage != nil {
+		preimage = *p.Preimage
+	}
+
+	return preimage
+}
+
+// ToMPPayment converts a legacy payment into an MPPayment.
```

Comment:
Confusion around the word legacy. In this context, it isn't a 'duplicate' payment is it?

Reply:
it's used for non-duplicate payments in this pr as well, but it's meant to refer to the legacy bucket structure. in the future, the non-duplicates will be migrated to the new mpp format, and then only the duplicates will continue using this conversion